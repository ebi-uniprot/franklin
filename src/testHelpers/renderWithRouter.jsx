import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react';

export default (
  ui,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {}
) => {
  // eslint-disable-next-line react/prop-types
  const Wrapper = ({ children }) => (
    <Router history={history}>{children}</Router>
  );
  return {
    ...render(ui, { wrapper: Wrapper }),
    // adding `history` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    history,
  };
};
