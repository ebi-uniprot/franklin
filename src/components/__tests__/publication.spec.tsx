import { screen, fireEvent } from '@testing-library/react';

import Publication from '../publication';

import publicationData from '../__mocks__/publications';

import renderWithRouter from '../../testHelpers/renderWithRouter';

let rendered: ReturnType<typeof renderWithRouter>;

describe('Publication component', () => {
  beforeEach(async () => {
    const {
      title,
      authors,
      abstract,
      infoData,
      journalInfo,
      pubmedId,
      statistics,
    } = publicationData;

    rendered = renderWithRouter(
      <Publication
        title={title}
        authors={authors}
        abstract={abstract}
        infoData={infoData}
        journalInfo={journalInfo}
        pubmedId={pubmedId}
        statistics={statistics}
        linkBuilder={(author) => `/citations/?query=author:"${author}"`}
      />
    );
  });

  test('should render', () => {
    const { asFragment } = rendered;
    expect(asFragment()).toMatchSnapshot();
  });

  test('should expand authors', async () => {
    expect(screen.queryByText('Darwin K.')).toBeNull();
    fireEvent.click(screen.getByText('[...]'));
    const author = await screen.findByText('Darwin K.');
    expect(author).toBeTruthy();
  });

  test('should expand abstract', async () => {
    expect(screen.queryByText(/Captain Fitzroy/)).toBeNull();
    fireEvent.click(screen.getByText('View abstract [...]'));
    const abstract = await screen.findByText(/Captain Fitzroy/);
    expect(abstract).toBeTruthy();
  });
});
