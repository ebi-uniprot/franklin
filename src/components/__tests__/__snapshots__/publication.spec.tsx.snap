// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Publication component should render 1`] = `
<DocumentFragment>
  <section
    class="publication"
  >
    <section
      class="publication__columns"
    >
      <section
        class="publication__columns__main"
      >
        <h5>
          The Zoology of the Voyage of H.M.S. Beagle.
        </h5>
        <section
          class="publication__authors"
        >
          <a
            href="/citations/?query=author:\\"Darwin C.\\""
          >
            Darwin C.
          </a>
          , 
          <a
            href="/citations/?query=author:\\"Darwin B.\\""
          >
            Darwin B.
          </a>
          , 
          <a
            href="/citations/?query=author:\\"Darwin A.\\""
          >
            Darwin A.
          </a>
          , 
          <a
            href="/citations/?query=author:\\"Darwin D.\\""
          >
            Darwin D.
          </a>
          , 
          <a
            href="/citations/?query=author:\\"Darwin E.\\""
          >
            Darwin E.
          </a>
          , 
          <a
            href="/citations/?query=author:\\"Darwin F.\\""
          >
            Darwin F.
          </a>
          , 
          <a
            href="/citations/?query=author:\\"Darwin G.\\""
          >
            Darwin G.
          </a>
          , 
          <a
            href="/citations/?query=author:\\"Darwin H.\\""
          >
            Darwin H.
          </a>
          , 
          <a
            href="/citations/?query=author:\\"Darwin I.\\""
          >
            Darwin I.
          </a>
          <button
            class="button tertiary"
            type="button"
          >
            [...]
          </button>
          , 
          <a
            href="/citations/?query=author:\\"Darwin X.\\""
          >
            Darwin X.
          </a>
        </section>
        <section
          class="publication__abstract"
        >
          <button
            class="button tertiary"
            type="button"
          >
            View abstract [...]
          </button>
        </section>
        <ul
          class="info-list"
        >
          <li>
            <div
              class="decorated-list-item decorated-list-item--compact"
            >
              <div
                class="decorated-list-item__title"
              >
                <h5
                  class="bold"
                >
                  Tortoise
                </h5>
              </div>
              <div
                class="decorated-list-item__content"
              >
                <em>
                  Reptile species of the family Testudinidae of the order Testudines (the turtles).
                </em>
              </div>
            </div>
          </li>
          <li>
            <div
              class="decorated-list-item decorated-list-item--compact"
            >
              <div
                class="decorated-list-item__title"
              >
                <h5
                  class="bold"
                >
                  Beetles
                </h5>
              </div>
              <div
                class="decorated-list-item__content"
              >
                Charles really liked them.
              </div>
            </div>
          </li>
        </ul>
      </section>
      <section
        class="publication__columns__side"
      >
        <section
          class="publication__columns__side__item"
        >
          <test-file-stub
            height="2em"
            width="1.875em"
          />
          <ul
            class="no-bullet"
          >
            <li>
              <a
                href="//www.ncbi.nlm.nih.gov/pubmed/123456"
                rel="noopener noreferrer"
                target="_blank"
              >
                PubMed
              </a>
            </li>
            <li>
              <a
                href="//europepmc.org/article/MED/123456"
                rel="noopener noreferrer"
                target="_blank"
              >
                Europe PMC
              </a>
            </li>
            <li>
              <small>
                <a
                  href="//dx.doi.org/10.1038/331525a0"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Nature 331:525-527(
                  <time
                    datetime="1988-01-01T00:00:00.000Z"
                  >
                    1988
                  </time>
                  )
                </a>
              </small>
            </li>
          </ul>
        </section>
        <section
          class="publication__columns__side__item"
        >
          <section
            class="publication__statistics"
          >
            <section
              class="publication__statistics__item"
            >
              <a
                href="/"
              >
                <section>
                  <small>
                    Mapped to
                  </small>
                </section>
                <section
                  class="publication__statistics__bubble"
                >
                  <span
                    class="bubble--small colour-pastel-blue"
                  >
                    3
                  </span>
                  <test-file-stub
                    height="15"
                    width="15"
                  />
                </section>
              </a>
            </section>
            <section
              class="publication__statistics__item"
            >
              <a
                href="/"
              >
                <section>
                  <small>
                    Cited in
                  </small>
                </section>
                <section
                  class="publication__statistics__bubble"
                >
                  <span
                    class="bubble--small colour-pastel-blue"
                  >
                    1
                  </span>
                  <test-file-stub
                    height="15"
                    width="15"
                  />
                </section>
              </a>
            </section>
          </section>
        </section>
      </section>
    </section>
  </section>
</DocumentFragment>
`;
